src/components/Image/RelativeMediaImage/RelativeMediaImage.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { MediaImage } from 'wix-ui-core/media-image';
+import { Dimensions, ImageProps } from '../types';
+import { classes, st } from './RelativeMediaImage.st.css';
+
+interface RelativeMediaImageProps extends Omit<ImageProps, 'width' | 'height'> {
+  sourceDimensions: Dimensions;
+  containerDimensions: Dimensions;
+  isPlaceholderDisplayed?: boolean;
+}
+
+const Placeholder = ({
+  src,
+  className,
+  sourceDimensions,
+  containerDimensions,
+  ...imageProps
+}: RelativeMediaImageProps) => (
+  <MediaImage
+    mediaPlatformItem={{
+      uri: src,
+      ...sourceDimensions,
+      options: {
+        filters: {
+          blur: 3,
+        },
+      },
+    }}
+    className={className}
+    {...containerDimensions}
+    {...imageProps}
+  />
+);
+
+export class RelativeMediaImage extends React.Component<RelativeMediaImageProps> {
+  render() {
+    const {
+      src,
+      className,
+      sourceDimensions,
+      containerDimensions,
+      isPlaceholderDisplayed,
+      onLoad,
+      ...imageProps
+    } = this.props;
+
+    return (
+      containerDimensions && (
```

Comment:
Actually without these dimensions we cannot render at all...

But we make sure that the component does get them (from fixed props/bounding rect). This condition is important because in case of the bounding rect, it happens in the didMount so we need to block the render until we have them (otherwise ui-core's MediaImage will fail to render).